yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -836,8 +866,8 @@ private[spark] class Client(
     val amArgs =
       Seq(amClass) ++ userClass ++ userJar ++ primaryPyFile ++ primaryRFile ++
         userArgs ++ Seq(
-          "--executor-memory", args.executorMemory.toString + "m",
-          "--executor-cores", args.executorCores.toString,
+          "--executor-memory", executorMemory.toString + "m",
```

Comment:
Do we need to pass these as explicit command line arguments to the AM? Since we're simplifying code, might just take the chance and make the AM read these from the configuration too.